pkg/oci/client/oci.go

```diff
@@ -431,3 +456,214 @@ func (c *client) WaitForVolumeDetached(volumeAttachmentId string) error {
+		return false, err
+	}
+	if exportSummary != nil {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *client) AttachFileSystemToMountTarget(fileSystem *ffsw.FileSystem, mountTarget *ffsw.MountTarget, path string) error {
+	exportSummary, err := c.findExport(fileSystem, mountTarget, path)
+	if err != nil {
+		return err
+	}
+	if exportSummary != nil {
+		log.Printf("Found export %s", *exportSummary.Id)
+		log.Printf("FileSystem:%s already mounted on MountTarget %s at %s", *fileSystem.Id, *mountTarget.Id, path)
+		return nil
+	}
+	response, err := func() (ffsw.CreateExportResponse, error) {
+		ctx, cancel := context.WithTimeout(c.ctx, c.timeout)
+		defer cancel()
+		return c.ffsw.CreateExport(ctx,
+			ffsw.CreateExportRequest{
+				CreateExportDetails: ffsw.CreateExportDetails{
+					ExportSetId:  mountTarget.ExportSetId,
+					FileSystemId: fileSystem.Id,
+					Path:         common.String(path),
+				},
+			})
+	}()
+	if err != nil {
+		return err
+	}
+	log.Printf("Filesystem Exported %s at %s(%s) %s", *fileSystem.Id, *mountTarget.Id, path, *response.Export.Id)
+
+	export := response.Export
+
+	for {
+		log.Printf("Export State:(%s)%s", *export.Id, export.LifecycleState)
+		if export.LifecycleState == ffsw.ExportLifecycleStateActive {
+			break
+		}
+
+		response, err := func() (ffsw.GetExportResponse, error) {
+			ctx, cancel := context.WithTimeout(c.ctx, c.timeout)
+			defer cancel()
+			return c.ffsw.GetExport(ctx, ffsw.GetExportRequest{
+				ExportId: export.Id,
+			})
+		}()
+		if err != nil {
+			return err
+		}
+		export = response.Export
+		time.Sleep(time.Second * 1)
+	}
+
+	return nil
+}
+
+func (c *client) DetachFileSystemToMountTarget(fileSystem *ffsw.FileSystem, mountTarget *ffsw.MountTarget, path string) error {
+	export, err := c.findExport(fileSystem, mountTarget, path)
+	if err != nil {
+		return err
+	}
+	if export == nil {
+		log.Printf("FileSystem:%s not mounted on MountTarget %s at %s", *fileSystem.Id, *mountTarget.Id, path)
+		return nil
+	}
+	log.Printf("Found export %s", *export.Id)
+
+	ctx, cancel := context.WithTimeout(c.ctx, c.timeout)
+	defer cancel()
+	_, err = c.ffsw.DeleteExport(ctx, ffsw.DeleteExportRequest{
+		ExportId: export.Id,
+	})
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Deleted export %s", *export.Id)
+
+	/*exportid := export.Id
+
+	for {
+		response, err := func() (ffsw.GetExportResponse, error) {
+			ctx, cancel := context.WithTimeout(c.ctx, c.timeout)
+			defer cancel()
+			return c.ffsw.GetExport(ctx, ffsw.GetExportRequest{
+				ExportId: exportid,
+			})
+		}()
+		if err != nil {
+			return err
+		}
+		log.Printf("Export State:(%s)%s", *export.Id, export.LifecycleState)
+		if export.LifecycleState == ffsw.ExportSummaryLifecycleStateDeleted {
+			break
+		}
+		time.Sleep(time.Second * 1)
+
+		exportid = response.Export.Id
+
+	}*/
+
+	return nil
+}
+
+func (c *client) GetMountTargetIPS(mountTarget *ffsw.MountTarget) ([]core.PrivateIp, error) {
+	var privateIps []core.PrivateIp
```

Comment:
`s/privateIps/privateIPs/`

Reply:
I'm matching the struct name so I disagree for consistancy!